ApplensBackend/Helpers/DateTimeHelper.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace AppLensV3.Helpers
+{
+    public class DateTimeHelper
+    {
+        internal static bool PrepareStartEndTimeWithTimeGrain(string startTime, string endTime, string timeGrain, int timeWindowInDays, out DateTime startTimeUtc, out DateTime endTimeUtc, out TimeSpan timeGrainTimeSpan, out string errorMessage)
+        {
+            timeGrainTimeSpan = KustoConstants.DefaultTimeGrain;
+            bool result = true;
+            errorMessage = string.Empty;
+
+            result = PrepareStartEndTimeUtc(startTime, endTime, timeWindowInDays, out startTimeUtc, out endTimeUtc, out errorMessage);
+
+            startTimeUtc = GetDateTimeInUtcFormat(RoundDownTime(startTimeUtc, timeGrainTimeSpan));
+            endTimeUtc = GetDateTimeInUtcFormat(RoundDownTime(endTimeUtc, timeGrainTimeSpan));
+
+            if (startTimeUtc == endTimeUtc)
+            {
+                endTimeUtc = startTimeUtc.Add(TimeSpan.FromDays(timeWindowInDays));
+            }
+
+            return result;
+        }
+
+        internal static bool PrepareStartEndTimeUtc(string startTime, string endTime, int timeWindowInDays, out DateTime startTimeUtc, out DateTime endTimeUtc, out string errorMessage)
+        {
+            //1. no startTime, no endTime => return current time - 24 hours, current time
+            //2. startTime, no endTime => return start time, end time = start time + 24 hours
+            //3. no startTime, endTime => return start time = end time - 24 hours, end time
+            //4. startTime, endTime => return start time, end time
+
+            DateTime currentUtcTime = GetDateTimeInUtcFormat(DateTime.UtcNow);
+            bool result = true;
+            errorMessage = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(startTime) && string.IsNullOrWhiteSpace(endTime))
+            {
+                endTimeUtc = currentUtcTime;
+                startTimeUtc = endTimeUtc.AddDays(-(timeWindowInDays));
```

Comment:
The default value for timeWindowInDays is 30?

Reply:
correct

Reply:
Ok.. Then why the comment above mentions the default timeWindow is 24 hours?

Reply:
if no start time and end time is specified, then we default to 24 hrs. for Outages, I specify the time period explicitly to be 30 days. (I was planning to build a comm history page for applens, but couldnt get to that)